From 3cbb5689713a5c9e7a552bed5841bf0ae5d904b6 Mon Sep 17 00:00:00 2001
From: Julien Vuillaumier <julien.vuillaumier@nxp.com>
Date: Fri, 17 Sep 2021 19:58:36 +0000
Subject: [PATCH] [armnn] adaptation for vsi armnn fork to support npu backend

Support for VSI NPU backend through TensorFilter element property:
  accelerator=true:npu

Upstream-Status: Inappropriate (depends on VSI patch in armnn component)

Signed-off-by: Julien Vuillaumier <julien.vuillaumier@nxp.com>
---
 ext/nnstreamer/tensor_filter/tensor_filter_armnn.cc | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ext/nnstreamer/tensor_filter/tensor_filter_armnn.cc b/ext/nnstreamer/tensor_filter/tensor_filter_armnn.cc
index 0fcc72f9..cfe79565 100644
--- a/ext/nnstreamer/tensor_filter/tensor_filter_armnn.cc
+++ b/ext/nnstreamer/tensor_filter/tensor_filter_armnn.cc
@@ -53,7 +53,7 @@
 
 static const gchar *armnn_accl_support[]
     = { ACCL_CPU_NEON_STR, /** ACCL for default and auto config */
-        ACCL_CPU_STR, ACCL_GPU_STR, NULL };
+        ACCL_CPU_STR, ACCL_GPU_STR, ACCL_NPU_STR, NULL };
 
 /**
  * @brief	ring cache structure
@@ -359,6 +359,8 @@ ArmNNCore::getBackend (const accl_hw hw)
   switch (hw) {
   case ACCL_GPU:
     return armnn::Compute::GpuAcc;
+  case ACCL_NPU:
+    return armnn::Compute::VsiNpu;
   case ACCL_NONE:
   /** intended */
   case ACCL_CPU:
-- 
2.25.1

